bench/wtperf/wtperf.c

```diff
@@ -2068,8 +2138,12 @@ main(int argc, char *argv[])
 	 * If the user wants compaction, then we also enable async for
 	 * the compact operation, but not for the workloads.
 	 */
-	if (cfg->async_threads > 0)
+	if (cfg->async_threads > 0) {
+		if (cfg->workload->truncate > 0) {
```

Comment:
If this is a single line, no need for curly braces. It would be better to output an error message for the user, telling them why we are erroring out.